newIDE/app/src/EventsFunctionsExtensionsLoader/MetadataDeclarationHelpers.js

```diff
@@ -590,18 +649,23 @@ export const declareEventsFunctionParameters = (
       } else {
         instructionOrExpression.addCodeOnlyParameter(
           parameter.getType(),
-          '' // See below for adding the extra information
+          parameter.getExtraInfo()
         );
       }
-      // Manually add the "extra info" without relying on addParameter (or addCodeOnlyParameter)
-      // as these methods are prefixing the value passed with the extension namespace (this
-      // was done to ease extension declarations when dealing with object).
-      instructionOrExpression
-        .getParameter(instructionOrExpression.getParametersCount() - 1)
-        .setExtraInfo(parameter.getExtraInfo());
     }
   );
 
+  const functionType = eventsFunction.getFunctionType();
+  if (functionType === gd.EventsFunction.ExpressionAndCondition) {
+    ((instructionOrExpression: any): gdMultipleInstructionMetadata).useStandardParameters(
+      'number'
+    );
+  } else if (functionType === gd.EventsFunction.StringExpressionAndCondition) {
+    ((instructionOrExpression: any): gdMultipleInstructionMetadata).useStandardParameters(
+      'string'
+    );
+  }
+
   // By convention, latest parameter is always the eventsFunctionContext of the calling function
   // (if any).
   instructionOrExpression.addCodeOnlyParameter('eventsFunctionContext', '');
```

Comment:
The operator and operand parameters are put at the end to avoid to mess withe the user defined parameter indexes that are used in sentense like "_PARAM3_".
The issue with this is that when a user add a new parameter, it breaks the compatibility. This should not be a huge issue as most ExpressionAndCondition won't have any extra parameters.